### PR TITLE
Merge code and history of GPOS into ORCA

### PR DESCRIPTION
**Pull request by d**
This PR merges both code and history of the GPOS library into ORCA repo.

We've changed

0. The build system of ORCA to
  0. build GPOS as part of ORCA
  0. run GPOS tests as part of ORCA tests
0. The ORCA pipeline to eliminate parts pertaining to GPOS
0. README
0. Bumped ORCA version from 1.697 to 2.0

To reviewers, the Github PR UI will have too much noise, it'll be far easier to `git fetch` our branch to see relevant commits in your local Git environment.

This pull request supersedes #130

### CODE REVIEW

**Comment by hsyuan** [comment]
+1

**Comment by dhanashreek89** [comment]
Looks like the `version.h.in` still exists underneath the `libgpos`, and should be removed also.

File: cmake/FindAtomics.cmake [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# Copyright (c) 2015, Pivotal Software, Inc.
```
Comment (vraghavan78) [inline cmake/FindAtomics.cmake]: Is this a new file?

File: libgpos/targzs/ivysettings.xml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (vraghavan78) [inline libgpos/targzs/ivysettings.xml]: do we need this?

File: libgpos/targzs/ivy-devel.xml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<ivy-module version="2.0">
```
Comment (vraghavan78) [inline libgpos/targzs/ivy-devel.xml]: do we need this?

File: cmake/FindAtomics.cmake (hunk 1 above)
Comment (hsyuan) [inline cmake/FindAtomics.cmake]: no

File: libgpos/targzs/ivy.xml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+<ivy-module version="2.0">
```
Comment (vraghavan78) [inline libgpos/targzs/ivy.xml]: do we need this>

File: libgpos/targzs/ivy-devel.xml (hunk 3 above)
Comment (hsyuan) [inline libgpos/targzs/ivy-devel.xml]: the whole folder can be removed.

File: cmake/FindAtomics.cmake (hunk 1 above)
Comment (d) [inline cmake/FindAtomics.cmake]: No venky, this is gotten from GPOS. We use this to detect support for atomic intrinsics. Currently GPOS uses a (somewhat portable, but less hip) implementation of atomics that's somehow tied to GCC (but it's also somewhat widely supported). This is necessary for our implementation of `gpos::UllExchangeAdd` and friends.

File: libgpos/targzs/ivysettings.xml (hunk 2 above)
Comment (d) [inline libgpos/targzs/ivysettings.xml]: will remove this entire folder

**Comment by d** [comment]
@hsyuan and I added two commits to address the above comments, specifically we

0. Removed the GPOS version, and with it, the generated version header
0. Removed unused artifact xml files

@vraghavan78 @xinzweb @dhanashreek89 please advice when you have time.

Jesse and Haisheng

**Pull request merged**